core/src/bundle_stage/bundle_reserved_space_manager.rs

```diff
@@ -186,4 +188,41 @@ mod tests {
             block_cost_limits
         );
     }
+
+    #[test]
+    fn test_block_limits_after_first_slot() {
+        const BUNDLE_BLOCK_COST_LIMITS_RESERVATION: u64 = 100;
+
+        let genesis_config_info = create_genesis_config(100);
+        let bank = Arc::new(Bank::new_for_tests(&genesis_config_info.genesis_config));
+
+        for _ in 0..genesis_config_info.genesis_config.ticks_per_slot {
+            bank.register_tick(&Hash::default());
+        }
+        assert!(bank.is_complete());
+        bank.freeze();
+
+        let bank1 = Arc::new(Bank::new_from_parent(bank, &Pubkey::default(), 1));
+
+        // Prints: bank1: 1, tick_height: 64, max_tick_height: 128
+        // println!(
+        //     "bank1: {}, tick_height: {}, max_tick_height: {}",
+        //     bank1.slot(),
+        //     bank1.tick_height(),
+        //     bank1.max_tick_height()
+        // );
+
+        let block_cost_limits = bank1.read_cost_tracker().unwrap().block_cost_limit();
+        let mut reserved_space = BundleReservedSpaceManager::new(
+            block_cost_limits,
+            BUNDLE_BLOCK_COST_LIMITS_RESERVATION,
+            5,
+        );
+        reserved_space.tick(&bank1);
```

Comment:
can you add a test case that tests the cost tracker reverting back to original block cost limits, after it's advanced enough ticks?